docs/API/API-functions.md

```diff
@@ -358,27 +358,74 @@ int srt_bind(SRTSOCKET u, const struct sockaddr* name, int namelen);
 
 Binds a socket to a local address and port. Binding specifies the local network
 interface and the UDP port number to be used for the socket. When the local 
-address is a form of `INADDR_ANY`, then it's bound to all interfaces. When the 
-port number is 0, then the port number will be system-allocated if necessary.
+address is a wildcard (`INADDR_ANY` for IPv4 or `in6addr_any` for IPv6), then
+it's bound to all interfaces (although see `SRTO_IPV6ONLY` and an additional
+information below for details about the wildcard address in IPv6).
+
+Binding is necessary for every socket to be used for communication. It can be
+bound implicitly, when the socket is being connected to a listening socket in
+the network using [`srt_connect`](#srt_connect) or a similar function that does
+the same, in which case the socket will be bound to a wildcard address of
+the IP version as specified in the `sockaddr::sa_family` field and port number
+0.  In other cases it must be bound explicitly using this function, or some
+other function that uses this functionality.
+
+When the port number is 0 (including the implicit binding case), then the port
+number will be system-allocated and can be determined after an explicit or
+implicit binding using [`srt_getsockname`](#srt_getsockname).
```

Comment:
I just wanted to mention here, should the reader have forgotten, that port 0 is also used when binding is explicit (that is, the `srt_connect` was called without calling `srt_bind` first), just as well it can be set this value when binding explicitly (by calling `srt_bind`).

Reply:
@ethouris Sorry, I'm not a port expert. When you say "when a port number is 0" followed by "the port number will be system allocated" are you referring to two different ports?

Reply:
With `srt_bind` you define what port you request from the system to allocate, and it will allocate, unless this port is busy. In this case, if the operation has succeeded, you can take as a good deal that the socket was bound to exactly that port.

But by specifying the port number 0 you request the system to find the first free port for you (so it will find, for example, 54321), so after `srt_bind` succeded you don't know which port the system has bound the socket to. To extract this information, you have to call `srt_getsockname`.

This rule for port number 0 is actually how the system does it, when SRT binds the underlying UDP socket to that address. SRT then calls the `getsockname` itself, after successful `bind`, to record this address in its internal data, and this is what the user's call to `srt_getsockname` will return.

Reply:
Maybe this:

```suggestion
When the port number specified for binding is 0 (including when binding was done implicitly), then the effective port number will be the one that system allocates automatically. To obtain this effective port number you can use [`srt_getsockname`](#srt_getsockname).
```

Reply:
Above is okay, but here is another possibility based on your earlier comments:

```suggestion
Port 0 is used when binding is implicit (that is, when `srt_connect` is called 
without calling `srt_bind` first). It can also be set to this value when binding 
explicitly (by calling `srt_bind`). With `srt_bind` you define what port you request 
from the system, and it will allocate it, unless this port is busy. In such cases, 
if the operation has succeeded, you can be certain that the socket was bound to 
the port you requested. But by specifying port number 0 (including when binding 
implicitly) you request the system to find the first available port. If `srt_bind` 
succeeds, you don't know to which port the system has bound the socket, since the 
effective port number will be the one that the system allocates automatically. To 
obtain this information, you can use [`srt_getsockname`](#srt_getsockname).
```